salt/utils/schedule.py

```diff
@@ -173,7 +174,11 @@ def _check_max_running(self, func, data, opts):
         data['run'] = True
         if 'jid_include' not in data or data['jid_include']:
             jobcount = 0
-            for job in salt.utils.minion.running(self.opts):
+            if self.opts['__role'] == 'master':
+                current_jobs = salt.utils.master.running(self.opts)
+            else:
+                current_jobs = salt.utils.minion.running(self.opts)
```

Comment:
This is not really nice approach, since e.g. Syndic is out here. Why not support structure like this:
```python
role = self.opts['__role']
if hasattr(self.utils, role):
    current_jobs = getattr(self.utils, role).running(self.opts)
```
Of course, the "role" should point in the object inside `salt/utils/__init__.py` then. But in that case everyone will easily add not yet implemented roles, for the same Syndic. Now you are just hard-coding things.

Reply:
:+1: